app/components/mappingsTable/mappingsTable.component.spec.js

```diff
@@ -0,0 +1,132 @@
+            $httpBackend.whenGET('components/indexMenu/indexMenu.html').respond();
+
+            $httpBackend.whenGET('/ws/rest/v1/conceptmaptype').respond({results: []});
+            $httpBackend.whenGET('/ws/rest/v1/conceptsource').respond({results: []});
+            $httpBackend.whenGET('/ws/rest/v1/concept/uuid/mapping?v=full').respond({results: []});
+
+
+            component = $componentController('mappingsTable', {scope: scope}, {conceptUuid: "uuid"});
+            $httpBackend.flush();
+            expect(component.mapTypes).toEqualData([]);
+            //only one default source to search all
+            expect(component.sources.length).toEqualData(1);
+            expect(component.mappings).toEqualData([]);
+        });
+
+        describe('component: mappingsTable, test functions', function(){
+            var mockMapping1, mockMapping2, mockMapping3, mockMappings, notificationMock;
+            notificationMock = testUtils.getNotificationMock();
+
+            beforeEach(function(){
+                mockMapping1 = {conceptReferenceTerm: {conceptSource: {uuid: "uuid"}, code: "code"}, conceptMapType: {uuid: "uuid"}, uuid: "uuid11" };
+                mockMapping2 = {conceptReferenceTerm: {uuid: "uuid", conceptSource: {uuid: "uuid"}}, conceptMapType: {uuid: "uuid"}, uuid: "uuid22" };
+                mockMapping3 = {conceptReferenceTerm: {uuid: "uuid", conceptSource: {}}, conceptMapType: {uuid: "uuid"}, uuid: "uuid33" };
+                mockMappings = [mockMapping1, mockMapping2, mockMapping3];
+
+
+                $httpBackend.whenGET('manifest.webapp').respond(500, "");
+                $httpBackend.whenGET(/translation.*/).respond();
+                $httpBackend.whenGET('components/indexMenu/indexMenu.html').respond();
+
+                $httpBackend.whenGET('/ws/rest/v1/conceptmaptype').respond({results: []});
+                $httpBackend.whenGET('/ws/rest/v1/conceptsource').respond({results: []});
+                $httpBackend.whenGET('/ws/rest/v1/concept/uuid/mapping?v=full').respond({results: []});
+
+
+                component = $componentController('mappingsTable',
+                    {scope: scope},
+                    {conceptUuid: "uuid"},
+                    {openmrsNotification: notificationMock});
+                $httpBackend.flush();
+            })
+
+
+            it("should divide mappings and store mappings to new reference terms at the end of array", function(){
+                component.mappings = mockMappings;
+                expect(component.mappings.indexOf(mockMapping1)).toEqualData(0);
+                component.divideMappings();
+                expect(component.mappings.indexOf(mockMapping1)).toEqualData(component.mappings.length-1);
+            })
+
+            it("mapping validation should return false if conceptSource uuid is not defined", function(){
+                var mock  = {conceptReferenceTerm: {conceptSource: {}}, conceptMapType: {uuid: "uuid"}, uuid: "uuid33" };
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2741260426/issues/source?bid=3196969&fileBranchId=3354756#l90)